src/main/user-impl/java/com/mysql/cj/jdbc/ha/ConnectionProxy.java

```diff
@@ -307,9 +303,9 @@ class JdbcInterfaceProxy implements InvocationHandler {
     }
 
     public synchronized Object invoke(Object proxy, Method method, Object[] args) throws Throwable {
-      if (METHOD_EQUALS.equals(method.getName())) {
-        // Let args[0] "unwrap" to its InvocationHandler if it is a proxy.
-        return args[0].equals(this);
+      final String methodName = method.getName();
+      if (isDirectExecute(methodName)) {
```

Comment:
Hey @karenc-bq can you help me understand this code? I think this code might be causing this issue: https://github.com/awslabs/aws-mysql-jdbc/issues/308 and I've made a PR to fix it in https://github.com/awslabs/aws-mysql-jdbc/pull/310

Reply:
@sergiyvamz maybe karenc has left bq? do you happen to have an opinion about this code?